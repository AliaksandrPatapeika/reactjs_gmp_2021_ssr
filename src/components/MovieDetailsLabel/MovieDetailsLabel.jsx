import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const MovieDetailsLabelWrapper = styled.div`
  margin-bottom: 4px;
  color: #f65261;
`;
const MovieDetailsValue = styled.span`
  color: #ffffff;
`;

const MovieDetailsLabel = ({label, value}) => (
  <MovieDetailsLabelWrapper>
    {label}
    {': '}
    <MovieDetailsValue>{value}</MovieDetailsValue>
  </MovieDetailsLabelWrapper>
);

MovieDetailsLabel.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
};

export default MovieDetailsLabel;
