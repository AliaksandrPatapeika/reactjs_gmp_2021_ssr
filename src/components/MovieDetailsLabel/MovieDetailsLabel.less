@import "../../less/variables";

.movieDetailsLabel {
	margin-bottom: 4px;
	color: @red;

	.movieDetailsValue {
		color: @white;
	}
}
