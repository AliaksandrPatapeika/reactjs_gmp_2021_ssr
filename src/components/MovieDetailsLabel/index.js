import MovieDetailsLabel from './MovieDetailsLabel';

export default MovieDetailsLabel;
