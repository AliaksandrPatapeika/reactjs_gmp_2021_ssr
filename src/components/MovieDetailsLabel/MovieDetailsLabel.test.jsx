import React from 'react';
import renderer from 'react-test-renderer';

import MovieDetailsLabel from './MovieDetailsLabel';

describe('MovieDetailsLabel', () => {
  test('renders correctly', () => {
    const movieDetailsLabel = renderer
      .create(<MovieDetailsLabel label="mock label" value="mock value" />)
      .toJSON();

    expect(movieDetailsLabel).toMatchSnapshot();
  });
});
