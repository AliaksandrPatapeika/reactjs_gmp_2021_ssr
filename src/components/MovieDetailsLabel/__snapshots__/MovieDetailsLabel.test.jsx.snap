// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsLabel renders correctly 1`] = `
<div
  className="MovieDetailsLabel__MovieDetailsLabelWrapper-sc-35caid-0 wEuzg"
>
  mock label
  : 
  <span
    className="MovieDetailsLabel__MovieDetailsValue-sc-35caid-1 cQRzbq"
  >
    mock value
  </span>
</div>
`;
